brave/brave-core
# Tie Clipboard Write permission to Read

Resolves https://github.com/brave/brave-browser/issues/25145

TODO:
- [ ] Browser tests
- [x] UI update for content settings in brave://settings/content/clipboard (get text approved)


## Screenshots of Settings Pages

![image](https://user-images.githubusercontent.com/5284154/189454326-98870a75-aab7-4139-99db-68ef9dd66e25.png)
**(Update text to say 'access')**

![image](https://user-images.githubusercontent.com/5284154/189454294-1668b8ca-1129-49a1-8ace-924d479cc61b.png)
**(Update text to say 'access')**

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

1. Go to https://async-clipboard-text.glitch.me/
2. Confirm that clicking on clicking on Copy results in a permission prompt.
3. Confirm that permission prompt mentions writing AND reading, not just "see text and images copied to clipboard".
4. Denying it prevents both Copy and Paste from working.
5. Approving makes both work.
6. The text in brave://settings/content/clipboard is updated to mention reading and writing, not just reading.
7. Adding a website to Allow or Deny lists on brave://settings/content/clipboard is respected.
